src/widget.ts

```diff
@@ -156,9 +157,33 @@ export const widget: WidgetHelpers | null = (() => {
           timelineSupport: true,
           useE2eForGroupCall: e2eEnabled,
           fallbackICEServerAllowed: allowIceFallback,
+          // XXX: The client expects the list of foci in its constructor, but we don't
+          // know this until we fetch the config file. However, we can't wait to construct
+          // the client object or we'll miss the 'capabilities' request from the host app.
+          // As of writing this, I have made the embedded widget client send the 'contentLoaded'
+          // message so that we can use the widget API in less racy mode, but we need to change
+          // element-web to use waitForIFrameLoad=false. Once that change has rolled out,
+          // we can just start the client after we've fetched the config.
+          foci: [],
         }
       );
-      const clientPromise = client.startClient().then(() => client);
+
+      const clientPromise = new Promise<MatrixClient>((resolve) => {
+        (async () => {
+          await Config.init();
```

Comment:
this seems to be redundant with what the initializer is doing? do we not have the initializer in embedded mode?

Reply:
Ah, sorry, added a comment

Reply:
This migh result un duplicated downloads of the json right?

Reply:
`Config.init` has a few lines that look like this:

```typescript
    if (Config.internalInstance?.initPromise) {
      return Config.internalInstance.initPromise;
    }
```

so that doesn't appear to be the case. If there's already a request promise in flight, it returns that same promise.